src/qibo/models/variational.py

```diff
@@ -659,3 +663,100 @@ def _loss(params, falqon, hamiltonian):
         final_loss = _loss(parameters, self, self.hamiltonian)
         extra = {"energies": energy, "callbacks": callback_result}
         return final_loss, parameters, extra
+
+
+class VariationalCircuit(Circuit):
+    """Circuit configuration for use cases with variational
+    circuit parameters of the Parameter class.
+
+    This circuit is symbolic and cannot perform calculations.
+    A specific backend has to be used for performing calculations.
+
+    Args:
+        nqubits (int): Total number of qubits in the circuit.
+        density_matrix (bool): If `True`, the circuit would evolve density matrices.
+            Defaults to ``False``.
+        accelerators (dict): Dictionary that maps device names to the number of times each
+            device will be used. Defaults to ``None``.
+    """
+
+    def __init__(self, nqubits, accelerators=None, density_matrix=False):
+        super().__init__(nqubits, accelerators, density_matrix)
+
+    def _get_initparams(self):
+        """Retrieve parameter values or objects directly from gates"""
+
+        params = []
+        for gate in self.queue:
+            if isinstance(gate, (gates.ParametrizedGate)):
+                try:
+                    params.append(gate.initparams)
+                except AttributeError:
+                    params.append(gate.parameters)
+
+        if isinstance(params[0], (float, int, tuple)):
+            params = self.get_parameters()
+            if isinstance(params[0], tuple):
+                params = np.array([val for t in params for val in t])
+
+        return params
+
+    def _get_train_params(self):
+        # for array
+        if isinstance(self.initparams, np.ndarray):
+            return self.initparams
+
+        # for Parameter objects
+        else:
```

Comment:
If you `return`ed before, you don't need an `else` branch.

https://pylint.pycqa.org/en/latest/user_guide/messages/refactor/no-else-return.html

Reply:
Corrected